sdk/client-core/core/src/main/java/io/clientcore/core/util/ClientLogger.java

```diff
@@ -582,48 +591,18 @@ static String writeJsonFragment(Map<String, Object> context) {
                 return "";
             }
 
-            StringBuilder formatter = new StringBuilder(context.size() * 20);
-
-            // Keep track of whether we've written a value yet so we don't write a trailing comma.
-            // The previous implementation would delete the trailing comma, but internally this causes StringBuilder to
-            // copy the entirety of the string to a new buffer, which is very expensive.
-            boolean firstValueWritten = false;
-            for (Map.Entry<String, Object> pair : context.entrySet()) {
-                if (firstValueWritten) {
-                    formatter.append(',');
-                } else {
-                    firstValueWritten = true;
-                }
-
-                writeKeyAndValue(pair.getKey(), pair.getValue(), formatter);
-            }
-
-            return formatter.toString();
-        }
+            int speculatedSize = context.size() * 20;
+            try (AccessibleByteArrayOutputStream outputStream = new AccessibleByteArrayOutputStream(speculatedSize);
+                 JsonWriter jsonWriter = JsonProviders.createWriter(outputStream)) {
+                jsonWriter.writeMap(context, JsonWriter::writeUntyped).flush();
 
-        private static void writeKeyAndValue(String key, Object value, StringBuilder formatter) {
-            formatter.append('"');
-            JSON_STRING_ENCODER.quoteAsString(key, formatter);
-            formatter.append("\":");
-
-            if (value == null) {
-                formatter.append("null");
-            } else if (isUnquotedType(value)) {
-                JSON_STRING_ENCODER.quoteAsString(value.toString(), formatter);
-            } else {
-                formatter.append('"');
-                JSON_STRING_ENCODER.quoteAsString(value.toString(), formatter);
-                formatter.append('"');
+                // Will have opening '{' and trailing '}' which we don't want to include.
```

Comment:
don't we need this for the json representation though?

Reply:
Diff isn't great here which makes this confusing. 

This is the method created the pre-JSONified global context which takes the form of `"key1":"value1",...`, as it gets injected as a list of key-value pairs (or a JSON fragment).